st2common/st2common/garbage_collection/executions.py

```diff
@@ -94,6 +104,23 @@ def purge_executions(logger, timestamp, action_ref=None, purge_incomplete=False)
     else:
         logger.info('Deleted %s liveaction objects' % (deleted_count))
 
+    # 3. Delete ActionExecutionOutputDB objects
+    to_delete_exection_ids = [str(execution_db.id) for execution_db in to_delete_execution_dbs]
+    output_dbs_filters = {}
+    output_dbs_filters['execution_id'] = {'$in': to_delete_exection_ids}
+
+    try:
+        deleted_count = ActionExecutionOutput.delete_by_query(**output_dbs_filters)
```

Comment:
Why not reuse purge_execution_output_objects from below?

Reply:
I agree - will make this change.

This was written before that function, first I didn't allow output objects to be purged independently from the execution ones.

Reply:
Actually, ignore that - can't make that change since the code doesn't do exactly the same thing.

The `purge_execution_output_objects` function itself purges based just on the timestamp and the `purge_executions` one purges output objects which belong to the executions which are to be purged.

In theory, the end result set should be similar, but it's safer this way to ensure consistency - so we don't have orphan output objects without executions laying around.